lib/docker-toolbelt.coffee

```diff
@@ -322,15 +323,49 @@ DockerToolbelt::createDeltaAsync = (src, dest, onProgress) ->
 			200: true,
 			404: 'no such image',
 			500: 'server error',
-	}
+
+	locals = {}
+
+	progressHandler = (e) ->
+		onProgress?(e)
+		return if locals.imageId?
+		return if not e.status?
+		if (match = /^Created delta: (sha256:\w+)$/.exec(e.status))
+			locals.imageId = match[1]
+
```

Comment:
With all the new code in this PR and other PRs in this "series"... The thought that occurs to me is that we should be writing the new code in Typescript instead of Coffeescript. The old / existing code could remain in Coffeescript, with new code written in Typescript.